src/rgw/rgw_admin.cc

```diff
@@ -5936,29 +5936,33 @@ int main(int argc, const char **argv)
 
   if (opt_cmd == OPT_LC_LIST) {
     formatter->open_array_section("lifecycle_list");
-    map<string, int> bucket_lc_map;
+    vector<cls_rgw_lc_entry> entries;
     string marker;
 #define MAX_LC_LIST_ENTRIES 100
     if (max_entries < 0) {
       max_entries = MAX_LC_LIST_ENTRIES;
     }
     do {
-      int ret = store->list_lc_progress(marker, max_entries, &bucket_lc_map);
+      int ret = store->list_lc_progress(marker, max_entries, &entries);
       if (ret < 0) {
         cerr << "ERROR: failed to list objs: " << cpp_strerror(-ret) << std::endl;
         return 1;
       }
-      map<string, int>::iterator iter;
-      for (iter = bucket_lc_map.begin(); iter != bucket_lc_map.end(); ++iter) {
+      vector<cls_rgw_lc_entry>::iterator iter;
+      for (iter = entries.begin(); iter != entries.end(); ++iter) {
         formatter->open_object_section("bucket_lc_info");
-        formatter->dump_string("bucket", iter->first);
-        string lc_status = LC_STATUS[iter->second];
+        formatter->dump_string("bucket", iter->bucket);
+        string lc_status = LC_STATUS[iter->status];
         formatter->dump_string("status", lc_status);
+        formatter->dump_stream("start_time") << utime_t(iter->start_time);
```

Comment:
should be able to use `encode_json()` instead, probably better using it instead of all the direct `formatter->dump()` calls